nixos/default.nix

```diff
@@ -150,35 +149,35 @@ in {
           TimeoutStartSec = 90;
           SyslogIdentifier = "hm-activate-${username}";
 
-          ExecStart = let
-            systemctl =
-              "XDG_RUNTIME_DIR=\${XDG_RUNTIME_DIR:-/run/user/$UID} systemctl";
-
-            sed = "${pkgs.gnused}/bin/sed";
-
-            exportedSystemdVariables = concatStringsSep "|" [
-              "DBUS_SESSION_BUS_ADDRESS"
-              "DISPLAY"
-              "WAYLAND_DISPLAY"
-              "XAUTHORITY"
-              "XDG_RUNTIME_DIR"
-            ];
-
-            setupEnv = pkgs.writeScript "hm-setup-env" ''
-              #! ${pkgs.runtimeShell} -el
-
-              # The activation script is run by a login shell to make sure
-              # that the user is given a sane environment.
-              # If the user is logged in, import variables from their current
-              # session environment.
-              eval "$(
-                ${systemctl} --user show-environment 2> /dev/null \
-                | ${sed} -En '/^(${exportedSystemdVariables})=/s/^/export /p'
-              )"
-
-              exec "$1/activate"
-            '';
-          in "${setupEnv} ${usercfg.home.activationPackage}";
+          ExecStart = ["${pkgs.coreutils}/bin/ln -s ${usercfg.programs.ssh.configPath} ${usercfg.home.homeDirectory}/.ssh/config"];
+        };
+      })) cfg.users;
+
+    systemd.user.services = mapAttrs' (_: usercfg:
+      let username = usercfg.home.username;
+      in nameValuePair ("home-manager-${utils.escapeSystemdPath username}") {
+        description = "Home Manager environment for ${username}";
+        wantedBy = [ "default.target" ];
+
+        # user units cannot depend on system units
+        # TODO: Insert in the script logic for waiting on the nix socket via dbus
+        # like https://github.com/mogorman/systemd-lock-handler
+        # wants = [ "nix-daemon.socket" ];
+        # after = [ "nix-daemon.socket" ];
+
+        environment = serviceEnvironment;
+
+        unitConfig.ConditionUser = username; # each user will execute only its own activation
```

Comment:
`serviceConfig.user` should also be set no?

Reply:
No, each user has its own systemd session that loads every user unit that has the correct ConditionUser
https://www.freedesktop.org/software/systemd/man/systemd.unit.html#ConditionUser=

Reply:
Also User does nothing for user services
https://www.freedesktop.org/software/systemd/man/systemd.exec.html#User=